src/providers.rs

```diff
@@ -45,6 +61,19 @@ pub fn get_default_providers() -> Vec<RegisterProviderArgs> {
     ]
 }
 
+pub fn find_provider(f: impl Fn(&Provider) -> bool) -> Option<Provider> {
+    PROVIDERS.with(|providers| {
+        let providers = providers.borrow();
+        Some(
+            providers
+                .iter()
```

Comment:
Wouldn't it be more efficient to use a hash map rather than iterating through the whole list (possibly twice)?

Reply:
Currently feels like a premature optimization (expecting maybe 10 providers overall), but it would make sense to eventually use a HashMap.